data/transactions/logic/eval_test.go

```diff
@@ -4034,14 +4043,18 @@ func TestAnyRekeyToOrApplicationRaisesMinAvmVersion(t *testing.T) {
 			expected := fmt.Sprintf("program version must be >= %d", cse.validFromVersion)
 			for v := uint64(0); v < cse.validFromVersion; v++ {
 				ops := testProg(t, source, v)
-				testAppBytes(t, ops.Program, ep, expected, expected)
+				if ep.isFull() {
+					testAppBytes(t, ops.Program, ep, expected, expected)
+				}
 				testLogicBytes(t, ops.Program, ep, expected, expected)
 			}
 
 			// Should succeed for all versions >= validFromVersion
 			for v := cse.validFromVersion; v <= AssemblerMaxVersion; v++ {
 				ops := testProg(t, source, v)
-				testAppBytes(t, ops.Program, ep)
+				if ep.isFull() {
+					testAppBytes(t, ops.Program, ep)
```

Comment:
We make stunted EvalParams when we detect that there are no app calls in the group.  So and non `isFull` ep cannot evaluate an app call.

Since these tests sometimes supply txgroups that don't have app calls, we get a stunted isFull that can't do an app eval.  That seems ok to me, as we still check the behaviour by running the apps as logicsigs.

Reply:
I'm good to resolve this thread.  I had originally thought we may wish to alter behavior, but I'm now on-board as is:  https://github.com/algorand/go-algorand/pull/4646#issuecomment-1279384731.